falcon/request.py

```diff
@@ -2334,6 +2334,90 @@ def get_param_as_json(
 
         return val
 
+    def get_param_as_dict(
+        self,
+        name: str,
+        required: bool = False,
+        deep_object: bool = False,
+        store: StoreArg = None,
+        default: Any | None = None,
+    ) -> Any:
+        """Retrieve a query parameter as a dictionary.
+
+        Supports OpenAPI's deep object style (`param[key]=value`)
+        and list-based key/value pairs (e.g., `param=a,b,c,d`).
+
+        Args:
+            name (str): Parameter name, case-sensitive (e.g, 'sort')
+
+        Keyword Args:
+            required (bool): Set to ``True`` to raise ``HTTPBadRequest``
+                instead of returning ``None`` when the parameter is not found
+                (default ``False``).
+            deep_object (bool): Set to True to use the deepObject (as in OAS)
+                format.
+            store: A ``dict``-like object in which to place the value
+                of the param, but only if the param is found (default ``None``).
+            default (any): If the param is not found returns the
+                given value instead of ``None``
+
+        Returns:
+            dict: The value of the param if it is found. Otherwise, returns
+            ``None`` unless required is ``True``.
+
+        Raises:
+            HTTPBadRequest: A required param is missing from the request, or
+                the value could not be parsed from the parameter.
+
+        .. versionadded:: 4.1.0
+        """
+
+        output: dict[str, Any] | None = None
+
+        if deep_object:
+            oc: dict[str, Any] = {}
+            for key, value in self._params.items():
+                if not key.startswith(f'{name}[') and key.endswith(']'):
```

Comment:
I guess we could optimize this a bit by caching the format, but it's likely not that important

Reply:
Another thing to try would be to benchmark this flow vs using a regex, but I doubt a regex would be faster.